src/pykeen/ablation/ablation.py

```diff
@@ -46,7 +47,7 @@ def ablation_pipeline(
     optuna_config: Optional[Mapping[str, Any]] = None,
     evaluator_kwargs: Optional[Mapping[str, Any]] = None,
     evaluation_kwargs: Optional[Mapping[str, Any]] = None,
-    directory: Optional[str] = None,
```

Comment:
still needs backwards compatibility for `str` everywhere

Reply:
oh i see that the `Path` name has been overridden... this is also misleading

Reply:
rename this to PathHint or PathType (or see if pathlib actually already defined this as a type hint somewhere)